themes/classic/templates/catalog/_partials/product-details.tpl

```diff
@@ -35,7 +35,7 @@
     {/if}
   {/block}
 
-  {block name='product_availability_date'}
+  {block name='product_availability_date' && ({$smarty.now|date_format:'%Y-%m-%d'} < $product.availability_date)}
```

Comment:
I guess you wanted to modify the next line. :)

Anyway, this kind of check should be done on another place, for instance in the presenter. This will allow us to keep the template as simple as possible.